plugins/analytics-module-snowplow/src/apis/implementations/AnalyticsApi/SnowplowAnalytics.ts

```diff
@@ -0,0 +1,130 @@
+        this.debounceTime = debounceTime;
+        
+        // create the Snowplow tracker
+        console.log("**********Setting up analytics (or not...)********");
+        if (this.enabled) {
+            console.log("**********Analytics enabled...********");
+
+            newTracker(trackerId, endpoint, {
+                appId: appId,
+                cookieLifetime: cookieLifetime,
+                platform: "web",
+                contexts: {
+                    webPage: true
+                },
+                plugins: [LinkClickTrackingPlugin(), SiteTrackingPlugin()]
+            });
+
+            enableActivityTracking({
+                minimumVisitLength: 30,
+                heartbeatDelay: 30
+            });
+        }
+    }
+  
+    static fromConfig(config: Config): SnowplowAnalytics {
+        const enabled = config.getBoolean('app.analytics.snowplow.enabled');
+        const baseUrl = config.getString('app.baseUrl');
+        const endpoint = config.getString('app.analytics.snowplow.collectorUrl');
+        const appId = config.getOptionalString('app.analytics.snowplow.appId') || 'Snowplow_standalone_OCIO' ;
+        const trackerId = config.getOptionalString('app.analytics.snowplow.trackerId') || 'rt';
+        const cookieLifetime = config.getOptionalNumber('app.analytics.snowplow.cookieLifetime') ?? 86400 * 548;
+        const debounceTime = config.getOptionalNumber('app.analytics.snowplow.debounceTime') ?? 3000;
+
+        return new SnowplowAnalytics({
+            enabled,
+            baseUrl,
+            trackerId,
+            endpoint,
+            appId,
+            cookieLifetime,
+            debounceTime
+        });
+    }
+    
+    captureEvent(event: AnalyticsEvent): void {
+        if (this.enabled) {
+            switch (event.action) {
+                case "search":
+                    this.captureSearch(event);
+                    break;
+                case "navigate":
+                    trackPageView();
+                    break;
+                case "click":
+                case "discover":
+                    this.trackClick(event);
+                    break;
+            }
+        }
+    }
+
+    private trackClick(event: AnalyticsEvent): void {
+        let to: string = event.attributes?.to as string ?? '';
+        const hasDomain = new RegExp(/[A-Za-z0-9-]{1,63}\.[A-Za-z]{2,6}/).test(to);
+        const isMailto = to.startsWith('mailto:');
+
+        // add the baseUrl to relative path links (this is largely to remain consistent with previous analytics)
+        if (!hasDomain && !isMailto) {
+            to = (to.startsWith('/'))? this.baseUrl + to : this.baseUrl + '/' + to;
+        }
+
+        trackLinkClick({ targetUrl: to, elementContent: event.subject });
+    }
+
+    private trackSearch(event: AnalyticsEvent): void {
+        // trim whitespace, split into non-empty terms
+        trackSiteSearch({ terms: event.subject.trim().split(" ").filter( t => t ) });
```

Comment:
I saw this when we were doing the demo but just want to confirm: this will make each work in the search string its own term, right? vs pushing the whole search string in as one search term? 

I was kind of anticipating the latter here (aka not the splitting) so curious what you're thinking the benefit of the split is

Reply:
Correct, each word in the string will be a search term. The Snowplow docs indicate that the `trackSiteSearch` method takes the search string as an array of search terms. 
Snowplow docs [here](https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/tracking-events/site-search/#event)
The quote from Dan on [DEVX-987](https://bcdevex.atlassian.net/jira/software/c/projects/DEVX/issues/DEVX-987) also references taking an array of terms.

I'm not sure if there's an advantage one way or the other, but opted to follow the docs.

Reply:
ok cool! that's fine if that's what it's expecting. we can always iterate with dan if the results are coming up weird...